Xamarin.Forms.Platform.UAP/WindowsBasePlatformServices.cs

```diff
@@ -117,9 +134,10 @@ public IIsolatedStorageFile GetUserStoreForApplication()
 		// "Each view or window has its own dispatcher. In assigned access mode, you should not use the MainView dispatcher, 
 		// instead you should use the CurrentView dispatcher." Checking to see if this isn't null (i.e. the current window is
 		// running above lock) calls through GetCurrentView(), and otherwise through MainView.
-		public bool IsInvokeRequired => LockApplicationHost.GetForCurrentView() != null
+		public bool IsInvokeRequired => Forms.Dispatchers.Count == 1
```

Comment:
For this code here can we just get rid of all the static checks and if there's only one dispatcher just use that dispatcher? We've been having issues getting the static methods to work in all cases

https://github.com/xamarin/Xamarin.Forms/pull/2526#issuecomment-385806329
https://github.com/xamarin/Xamarin.Forms/issues/1790

Reply:
We are now storing all dispatchers in Forms.Dispatchers variable, and now the IsInvokeRequired first checks if there is more than one dispatcher. If there is only one, we return false, otherwise we continue with the other validations. Hopefully it can help solving the issues you mentioned, it would require more some testing though.

Reply:
> If there is only one, we return false

So is that logic not checked in yet? Or did I misunderstand you? 

My thinking is to maybe just setup IsInvokeRequired as
```C#
public bool IsInvokeRequired => 
				 Forms.Dispatchers.Count == 1
				?   !_dispatcher.HasThreadAccess //or retrieve the first dispatcher in the list
			: true; 
```

My main point is that if the code in the following form exists
```C#
LockApplicationHost.GetForCurrentView() != null
			? !CoreApplication.GetCurrentView().Dispatcher.HasThreadAccess
			: !CoreApplication.MainView.CoreWindow.Dispatcher.HasThreadAccess
```

It's going to cause issues either when the UWP app is a share target or with our Animation code

Reply:
Awesome!! 👍 

For reference here are the two issues we've been contending with trying to make all scenarios happy
https://github.com/xamarin/Xamarin.Forms/issues/2327
https://github.com/xamarin/Xamarin.Forms/issues/1790
https://github.com/xamarin/Xamarin.Forms/pull/1534

If you don't find any side-effects I'll test as well with the next commit 

In the case of a single dispatcher I've tested
```c#
public bool IsInvokeRequired =>  !_dispatcher.HasThreadAccess;
```

And the use cases I was able to throw at it seemed fine, so I'm hopeful.

Reply:
@edguer Can we get this rebased now that #2526 is in master? That PR eliminates the LockApplicationHost stuff, so I think you can go with the version of IsInvokeRequired that @PureWeen suggests upthread.